perun-core/src/main/java/cz/metacentrum/perun/core/impl/Auditer.java

```diff
@@ -432,7 +433,7 @@ public void initialize() throws InternalErrorException {
 			//Update list of already resolved messages
 			alreadyResolvedMessages.addAll(addedResolvedMessages);
 			//Continue of processing newly generated messages
-			addedResolvedMessages.addAll(checkRegisteredAttributesModules(session, addedResolvedMessages, alreadyResolvedMessages));
+			addedResolvedMessages.addAll(checkRegisteredAttributesModules(session, new ArrayList<>(addedResolvedMessages), alreadyResolvedMessages));
```

Comment:
Just a thought. Could we change the type of the argument for the method `checkRegisteredAttributesModules` from `List<AuditerMessage> messages` to `Collection<AuditerMessage> messages` ? With that, it would not be necessary to convert the LinkedHashSet to ArrayList.

Reply:
Fixed.